config.default

```diff
@@ -0,0 +1,75 @@
+{
+  "common": {
+    "sync_on_start": false
```

Comment:
> I would like to see how the service design ensures reliability. I would look at different point when the service may be hardly stopped.
> 
> 1. When several files are uploaded, but some are not.

I plan to create a ticket to make the file upload transactional (either everything was loaded or nothing).

> 2. When a file is partially uploaded (BTW, whether it'll be considered as existent by S3?).

As far as I understand, the file can't be partially uploaded, it is either uploaded or not.

> 3. When all files are uploaded, but metainfo is not updated.

This is a standard situation, metainformation is updated  later from the `sync` thread.

> 4. When all files are uploaded, but metainfo is partially updated (what states are possible here?)

`mkrepo` is responsible for consistently updating metainformation(in my mind).

> Since you likely investigate all or some cases, it would be nice to add them to readme.

I plann to do it a little bit later, because I haven't investigate it yet).

Reply:
Thanks for the answers! I'll leave the discussion open as reminder.

Reply:
https://github.com/tarantool/rws/issues/2